src/System.Numerics.Vectors/src/System/Numerics/Vector2.cs

```diff
@@ -244,14 +244,14 @@ public static Vector2 Clamp(Vector2 value1, Vector2 min, Vector2 max)
             // This compare order is very important!!!
             // We must follow HLSL behavior in the case user specified min value is bigger than max value.
             float x = value1.X;
-            x = (x > max.X) ? max.X : x;
-            x = (x < min.X) ? min.X : x;
+            x = (min.X > x) ? min.X : x;  // max(x, minx)
+            x = (max.X < x) ? max.X : x;  // min(x, maxx)
 
             float y = value1.Y;
-            y = (y > max.Y) ? max.Y : y;
-            y = (y < min.Y) ? min.Y : y;
-
-            return new Vector2(x, y);
+            y = (min.Y > y) ? min.Y : y;  // max(y, miny)
+            y = (max.Y < y) ? max.Y : y;  // min(y, maxy)
+            
```

Comment:
nit: Stray whitespace on this line and on https://github.com/dotnet/corefx/pull/32556/files#diff-1014c829686d8f8dba9a11b0019820b9R283